### PR TITLE
[Sign-in Revamp]: EVM wallets submodal implementation in Auth Modal

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6387

## Description of Changes
- Adds new sub-modal in AuthModal for EVM wallets (wallet connect, coinbase, metamask) + their integrations
- Adds community-specific wallet integrations (as it was in old modal)
  - In 'Axie-Infinity', only `Ronin` wallet option is available
  - In 'Terra', only `StationWallet` and `Terra-WalletConnect` (wallet connect for terra) options are available
  - In 'Near', only `Near` wallet option is available
  - In 'Evmos' / 'Injective', show `cosm-metamask` (metamask for cosmos, it has the same button as metamask wallet, but it works for cosmos community) and `keplr-ethereum` wallets (it has the same button as Keplr wallet)

## "How We Fixed It"
N/A

## Test Plan
- Enable `FLAG_NEW_SIGN_IN_MODAL` in `.env`
- Install and setup EVM wallets (metamask/coinbase).
- Logout and visit any non-community page or any Ethereum-based community.
- Open the AuthModal, and click on the `Wallet connect` option to open the EVM wallets submodal.
- From the list in submodal, click on any wallet (`Coinbase` might not work in local) and try to login, verify it logs you in correctly and closes both modals.
- Logout and try to login with `WalletConnect` (you might have to install some wallet on mobile (if testing on mobile), i tried with `Zerion`) from the evm wallets submodal, while it is logging in
  - verify you see the reset wallet connect button (it will not work in local)
  - verify it logs you in correctly.
image

- Visit the `/axie-infinity` community, open the auth modal and verify you see the `Ronin` wallet option if you have that wallet installed, else verify you see "No wallets available" message. Verify that logging in with that wallets works.
- Visit the `/terra` community, open the auth modal and verify you see the `StationWallet` wallet option if you have that wallet installed and you see the 'WalletConnect (Terra)' option (this is different from the general wallet connect flow), else verify you see "No wallets available" message. Verify that logging in with these wallets works.
- Visit the `/near` community, open the auth modal and verify you see the `Near` wallet option. Verify that logging in with this wallet works.

## Deployment Plan
Part of https://github.com/hicommonwealth/commonwealth/milestone/38

## Other Considerations
`FLAG_NEW_SIGN_IN_MODAL` should not be enabled in prod yet.

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/AuthButton/AuthButton.scss [hunk 1]
```diff
@@ -5,29 +5,63 @@
...
+
+    &:hover {
+      background-color: $neutral-25;
+    }
   }
 
-  &:hover {
-    background-color: $neutral-25;
+  &.dark {
+    background-color: #1e1f1f;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/AuthButton/AuthButton.scss:41]: can you use color from pallete

File: packages/commonwealth/client/scripts/views/components/AuthButton/AuthButton.tsx [hunk 2]
```diff
@@ -21,7 +24,7 @@ const AuthButton = ({
     <button
       disabled={disabled}
       onClick={disabled ? null : onClick}
-      className={clsx('AuthButton', className)}
+      className={clsx('AuthButton', variant, rounded && 'rounded', className)}
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/AuthButton/AuthButton.tsx:27]:
```suggestion
      className={clsx('AuthButton', variant, className, { rounded })}
```

File: packages/commonwealth/client/scripts/views/modals/AuthModal/EVMWalletsSubModal/EVMWalletsSubModal.scss [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  background-color: black;
+
+  .CWModal {
+    border-radius: 24px;
+    box-shadow: none;
+    background: transparent;
+  }
+
+  .container {
+    background-color: #191a1a;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/AuthModal/EVMWalletsSubModal/EVMWalletsSubModal.scss:13]: color?

**Review by masvelio — APPROVED** [review]
1. should something happen on hover/click question mark icon? Nothing happens now
image

2. not sure if it was like this or it got messed up recently but when you click EVM you should not see keplr and leap. If you click cosmos, you should not see evm related wallets. 

https://github.com/hicommonwealth/commonwealth/assets/14819225/8c9e797f-43e5-436c-a3dd-c917073564d2


Apart from these small comments, looks and works great!

File: packages/commonwealth/client/scripts/views/components/AuthButton/AuthButton.tsx (hunk 2 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/components/AuthButton/AuthButton.tsx:27]: fixed

File: packages/commonwealth/client/scripts/views/modals/AuthModal/EVMWalletsSubModal/EVMWalletsSubModal.scss (hunk 3 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/modals/AuthModal/EVMWalletsSubModal/EVMWalletsSubModal.scss:13]: discussing this [here](https://github.com/hicommonwealth/commonwealth/pull/6577#discussion_r1479793209)

File: packages/commonwealth/client/scripts/views/components/AuthButton/AuthButton.scss (hunk 1 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/components/AuthButton/AuthButton.scss:41]: this was a 1-off color usage, not sure if we want to add it to global color, but if yes, plz lmk.

**Comment by mzparacha** [comment]
This PR needs to be rebased on `malik.6384.sign-in-modal-revamp`, after that we can test.

**Comment by mzparacha** [comment]
PR rebased, good to test.

**Review by masvelio — APPROVED** [review]

**Review by kurtisassad — APPROVED** [review]

**Pull request merged**